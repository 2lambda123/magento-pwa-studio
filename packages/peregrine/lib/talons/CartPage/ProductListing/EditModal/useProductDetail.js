const stockStatusLabels = new Map([
    ['IN_STOCK', 'In stock'],
    ['OUT_OF_STOCK', 'Out of stock']
]);

export const useProductDetail = props => {
    const { item } = props;
    const { prices, product, quantity } = item;
    const { price } = prices;
    const { currency, value: unitPrice } = price;
    const {
        name,
        sku,
        small_image: smallImage,
        stock_status: stockStatusValue
    } = product;
    const { url: imageURL } = smallImage;

    return {
        currency,
        imageURL,
        name,
        quantity,
        unitPrice,
        sku,
        stockStatus: stockStatusLabels.get(stockStatusValue) || 'Unknown'
    };
};
