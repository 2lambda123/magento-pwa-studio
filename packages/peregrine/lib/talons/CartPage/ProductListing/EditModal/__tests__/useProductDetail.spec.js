import { useProductDetail } from '../useProductDetail';

const mockProduct = {
    name: 'Simple Product',
    sku: 'SP01',
    small_image: {
        url: 'https://test.cdn/image.jpg'
    }
};

const mockItem = {
    prices: {
        price: {
            currency: 'USD',
            value: 123.45
        }
    },
    quantity: 5
};

test('returns known stock status with all props', () => {
    const item = {
        ...mockItem,
        product: {
            ...mockProduct,
            stock_status: 'IN_STOCK'
        }
    };

    const talonProps = useProductDetail({ item });
    expect(talonProps).toMatchSnapshot();
});

test('returns unknown stock status', () => {
    const item = {
        ...mockItem,
        product: {
            ...mockProduct,
            stock_status: 'WOOF_WOOF'
        }
    };

    const talonProps = useProductDetail({ item });
    expect(talonProps.stockStatus).toEqual('Unknown');
});
