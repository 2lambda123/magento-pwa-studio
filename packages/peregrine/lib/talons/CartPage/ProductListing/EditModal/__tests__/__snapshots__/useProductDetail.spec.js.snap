// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns known stock status with all props 1`] = `
Object {
  "currency": "USD",
  "imageURL": "https://test.cdn/image.jpg",
  "name": "Simple Product",
  "quantity": 5,
  "sku": "SP01",
  "stockStatus": "In stock",
  "unitPrice": 123.45,
}
`;
